allocation.md

```diff
@@ -1,53 +1,406 @@
-  pvCost: .70                            // total cost of persistent volume allocated
-  ramByteHours: 123684420.26             // cumulative hours of RAM consumed, in bytes
-  ramCost: 0.023                         // total cost of RAM allocated
-  ramEfficiency: .56                     // percentage of RAM requested that is utilized, weighted by cost
-  sharedCost: 0
-  start: "2020-09-13T00:00:00+01:00"     // beginning of window
-  totalCost: 0.024                       // sum of all costs
-  totalEfficiency: .72                   // percentage of RAM + CPU equested that is utilized, weighted by cost
+# Allocation
+
+The Allocation API is the preferred way to query for costs and resources allocated to Kubernetes workloads, and optionally aggregated by Kubernetes concepts like `namespace`, `controller`, and `label`. Data is served from one of [Kubecost's ETL pipelines](https://github.com/kubecost/docs/blob/master/allocation-api.md#caching-overview). The endpoint is available at the URL:
+```
+http://<kubecost>/model/allocation
+```
+
+> **NOTE**
+>
+> Throughout, we use `localhost:9090` as the default Kubecost URL, but your Kubecost instance may be exposed by a service or ingress. To reach Kubecost at port 9090, run: `kubectl port-forward deployment/kubecost-cost-analyzer -n kubecost 9090`
+
+## Quick start
+
+Request allocation data for each 24-hour period in the last three days, aggregated by namespace:
+```
+$ curl http://localhost:9090/model/allocation \
+  -d window=3d \
+  -d aggregate=namespace \
+  -d accumulate=false \
+  -d shareIdle=false
+  -G
+```
+```json
+{
+  "code": 200,
+  "data": [
+    {
+      "__idle__": { ... },
+      "default": { ... },
+      "kube-system": { ... },
+      "kubecost": { ... }
+    },
+    {
+      "__idle__": { ... },
+      "default": { ... },
+      "kube-system": { ... },
+      "kubecost": { ... }
+    },
+    {
+      "__idle__": { ... },
+      "default": { ... },
+      "kube-system": { ... },
+      "kubecost": { ... }
+    },
+    {
+      "__idle__": { ... },
+      "default": { ... },
+      "kube-system": { ... },
+      "kubecost": { ... }
+    }
+  ]
+}
+```
+
+Note: querying for "3d" will likely return a range of four sets because the queried range will overlap with four precomputed 24-hour sets, each aligned to the configured timezone. For instance, querying "3d" on 2021/01/04T12:00:00 will return:
+- 2021/01/04 00:00:00 until 2021/01/04T12:00:00 (now)
+- 2021/01/03 00:00:00 until 2021/01/04 00:00:00
+- 2021/01/02 00:00:00 until 2021/01/03 00:00:00
+- 2021/01/01 00:00:00 until 2021/01/02 00:00:00
+
+See [Querying](#querying) for the full list of arguments and [Examples](#examples) for more example queries.
+
+## Allocation schema (version 1.76)
+Field | Description
+---: | :---
+name | Name of each relevant Kubernetes concept described by the allocation, delimited by slashes, e.g. "cluster/node/namespace/pod/container"
```

Comment:
Maybe note that the schema differs by aggregation or even include the different schemas for each aggregation (though we may not want to commit to them).

Reply:
I suppose `properties` should be the canonical location for this information.

Reply:
Right. As we've talked about, it might even be nice to deprecate name in favor is _just_ properties once the `Properties` struct is rewritten.

Reply:
This was probably the hardest one to define, for this reason.

Reply:
Looks good as-is then. We don't have to commit to a schema if we're going to deprecate it soon-ish (hopefully :smile:).